system/lib/libc/musl/src/internal/pthread_impl.h

```diff
@@ -150,4 +171,11 @@ void __restore_sigs(void *);
 #define DEFAULT_STACK_SIZE 81920
 #define DEFAULT_GUARD_SIZE PAGE_SIZE
 
+#define __ATTRP_C11_THREAD ((void*)(uintptr_t)-1)
+
+#if !__EMSCRIPTEN_PTHREADS__
+void __emscripten_init_pthread_stub(void);
```

Comment:
Turns out we weren't initializing locale in non-stub versions. Added a browser test to verify that both stub and non-stub versions have nonzero locale fields, and made the initialization consistent here https://github.com/kripken/emscripten/pull/4813/commits/d01f14d2115d365d6d11ffef9ca3e1357885529b